src/hash-table.h

```diff
@@ -80,6 +81,16 @@ typedef void *HashTableKey;
 typedef void *HashTableValue;
 
 /**
+ * Internal structure representing an entry in hash table
+ * used as @ref HashTableIterator next result.
+ */
+
+typedef struct _KeyValuePair{
```

Comment:
I've tried to make type+function names namespaced by data structure, eg. HashTableIterator. Can you rename this something like "HashTablePair"?